src/compiler/parser.ts

```diff
@@ -1333,58 +1334,80 @@ module ts {
         }
 
         function parseComputedPropertyName(): ComputedPropertyName {
-            // PropertyName[Yield,GeneratorParameter] :
+            // PropertyName[Yield,GeneratorParameter,Await,AsyncParameter] : (Modified)
             //     LiteralPropertyName
             //     [+GeneratorParameter] ComputedPropertyName
-            //     [~GeneratorParameter] ComputedPropertyName[?Yield]
+            //     [+AsyncParameter] ComputedPropertyName
+            //     [~GeneratorParameter,~AsyncParameter] ComputedPropertyName[?Yield,?Await]
             // 
-            // ComputedPropertyName[Yield] :
-            //     [ AssignmentExpression[In, ?Yield] ]
+            // ComputedPropertyName[Yield,Await] :
+            //     [ AssignmentExpression[In,?Yield,?Await] ]
             //
             var node = <ComputedPropertyName>createNode(SyntaxKind.ComputedPropertyName);
             parseExpected(SyntaxKind.OpenBracketToken);
 
+            var explicitFlags = ParserContextFlags.None;
+            var ambientFlags = ParserContextFlags.None;
+            if (!inGeneratorParameterContext() && !inAsyncParameterContext()) {

```

Comment:
Yes, but this doesn't seem to account for the case where you are in exactly one of them. You only do something if you are in neither of them.